src/test/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporterTest.java

```diff
@@ -0,0 +1,643 @@
+
+        //then
+        assertTrue(result.failed());
+        assertThat(result.cause())
+                .hasMessageStartingWith("Error decoding imp.ext.prebid: "
+                        + "Cannot construct instance of `org.prebid.server.proto.openrtb.ext.request.ExtOptions`");
+    }
+
+    private static AuctionContext givenAuctionContext(Imp imp) {
+        // bid request
+        final Site site = Site.builder()
+                .domain("www.leparisien.fr")
+                .build();
+
+        final BidRequest bidRequest = BidRequest.builder()
+                .id("request1")
+                .imp(Collections.singletonList(imp))
+                .site(site)
+                .ext(givenExtRequest())
+                .build();
+
+        // bid response
+        final Bid bid = Bid.builder()
+                .id("bid1")
+                .impid("adunitcodevalue")
+                .price(BigDecimal.valueOf(1.5))
+                .adm("<div>Ad Markup</div>")
+                .build();
+
+        final SeatBid seatBidWithBid = SeatBid.builder()
+                .bid(Collections.singletonList(bid))
+                .seat("seat1")
+                .build();
+
+        final BidResponse bidResponse = BidResponse.builder()
+                .id("response1")
+                .seatbid(Collections.singletonList(seatBidWithBid))
+                .cur("USD")
+                .build();
+
+        final BidRejectionTracker bidRejectionTracker = new BidRejectionTracker(
+                "seat2",
+                Set.of("adunitcodevalue"),
+                1.0);
+
+        bidRejectionTracker.reject("imp1", BidRejectionReason.NO_BID);
+
+        return AuctionContext.builder()
+                .httpRequest(HttpRequestContext.builder().build())
+                .bidRequest(bidRequest)
+                .bidResponse(bidResponse)
+                .bidRejectionTrackers(
+                        singletonMap(
+                                "seat2",
+                                bidRejectionTracker))
+                .build();
+    }
+
+    private static Banner givenBanner() {
+        final Format format = Format.builder()
+                .w(320)
+                .h(50)
+                .build();
+
+        return Banner.builder()
+                .format(Collections.singletonList(format))
+                .w(240)
+                .h(400)
+                .build();
+    }
+
+    private static Banner givenBannerWithoutFormat() {
+        return Banner.builder()
+                .pos(1)
+                .format(null)
+                .w(728)
+                .h(90)
+                .build();
+    }
+
+    private static Video givenVideo() {
+        return Video.builder()
+                .pos(1)
+                .plcmt(1)
+                .build();
+    }
+
+    private static AuctionContext givenAuctionContextWithNoBidResponse() {
+        // bid request
+        final Site site = Site.builder()
```

Comment:
Separate methods for site, request, and context are better. 
it will be consistent with other methods like `givenVideo `.
Try to use customizers(UnaryOperator)